### ISSUE TITLE
Is there a chance MySQL OID allocation can produce duplicates?

### ISSUE THREAD

**Issue by jamadden** [zodb/relstorage#283]
Report from a test site running 3.0a5:

```python traceback
 File "/home//buildout/eggs/transaction-2.4.0-py2.7.egg/transaction/_transaction.py", line 421, in _commitResources
   rm.commit(self)
 File "/home//buildout/eggs/ZODB-5.5.1-py2.7.egg/ZODB/Connection.py", line 497, in commit
   self._commit(transaction)
 File "/home//buildout/eggs/ZODB-5.5.1-py2.7.egg/ZODB/Connection.py", line 546, in _commit
   self._store_objects(ObjectWriter(obj), transaction)
 File "/home//buildout/eggs/ZODB-5.5.1-py2.7.egg/ZODB/Connection.py", line 609, in _store_objects
   self._cache[oid] = obj
ValueError: A different object already has the same oid
```

It appeared to be transient; a retry succeeded.

**Comment by jamadden** [comment]
I haven't been able to produce anything of this nature. But I did produce some sort of issue [under PyPy (2 and 3)](https://travis-ci.org/zodb/relstorage/jobs/565540825#L508):
```python traceback
  Set up relstorage.tests.testmysql.MySQLPyMySQLConnector_Python in 0.000 seconds.
Traceback (most recent call last):
  File "/opt/python/pypy2.7-7.1.1/lib-python/2.7/threading.py", line 820, in __bootstrap_inner
    self.run()
  File "/opt/python/pypy2.7-7.1.1/lib-python/2.7/threading.py", line 773, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/travis/build/zodb/relstorage/src/relstorage/tests/reltestbase.py", line 584, in allocate_oids
    for _ in range(oid_count))
  File "/home/travis/build/zodb/relstorage/src/relstorage/tests/reltestbase.py", line 584, in <genexpr>
    for _ in range(oid_count))
  File "/home/travis/build/zodb/relstorage/src/relstorage/storage/__init__.py", line 503, in new_oid
    return self._oids.new_oid(bool(self._tpc_phase))
  File "/home/travis/build/zodb/relstorage/src/relstorage/storage/oid.py", line 91, in new_oid
    return int64_to_8bytes(oid_int)
  File "/home/travis/virtualenv/pypy2.7-7.1.1/site-packages/ZODB/utils.py", line 95, in p64
    return _OID_PACK(v)
ValueError: cannot convert negative integer to unsigned
```

**Comment by jamadden** [comment]
Ahh, the PyPy issue is that [same JIT bug](https://github.com/zodb/relstorage/issues/228#issuecomment-492423284); the same workaround appears to work.

**Comment by jamadden** [comment]
Fixed in #295
